drafts/25-xxx-specifications.txt

```diff
@@ -413,22 +413,141 @@ in28. Unlike INCLUDE lines (see the section on "INCLUDE lines"),
+      #else directive (if present) are part of the same list of
+      directives introduced by the #if directive.
+
+if20. The chain of conditional directives ends with the #endif
+      directive.
+
+if25. Within a conditional directive list, a #if, #ifdef, or #endif
+      directive introduces a new list of conditional directives, at a new
+      nesting level, within the enclosing conditional directive list.
+
+if30. The conditional directive lists must properly nest.
+
+if35. As shown in the illustrative syntax, there should be no
+      unmatched #endif directives.
+
+if40. As shown in the illustrative syntax, there should be no more
+      than one #else directive in a list of conditional directives at
+      any nesting level.
 
 
 3.5.2 Evaluation semantics specifications
 -----------------------------------------
 
 
+ix05. The #ifdef and #ifndef directives are evaluated as if they had been
+      written "#if defined(ID)" and "#if !defined(ID)" respectively.
+      (For brevity, the descriptions of #ifdef and #ifndef directives
+      below are omitted, and assume this transformation.)
+
+ix10. The #elifdef and #elifndef macros are evaluated as if they had
+      been written "#elif defined(ID)" and "#elif !defined(ID)"
+      respectively.
+
+ix15. The token-lists in #if and #elif are processed as described in
+      the section "Macro recognition and expansion".
+
+ix20. After expansion, the resulting token list in #if and #elif
+      directives must represent a valid integer expression as
+      described in the section "Expressions allowed in #if and #elif
+      directives". This expression is called the "controlling
+      expression" for the directive.
+
+ix25. Controlling expressions in #if and #elif directives are parsed
+      and evaluated according to the semantic rules described in
+      section "Expressions allowed in #if and #elif directives".
+
+ix30. If the controlling expression in a #if evaluates to a nonzero
+      value, the <lines> immediately following the #if participate in
+      preprocessing. They are subject to preprocessing expansion,
+      substitution, and evaluation of processing directives in the
+      <lines>.
+
+ix40. Further preprocessing when #if controlling expression evaluates
+      to a nonzero value proceeds. Subsequent #elif and #else
+      directives at the same level of nesting are skipped, and do not
+      participate in preprocessing except to be examined for proper
+      nesting of conditional directives.
+
+ix45. If the controlling expression in a #if evaluates to a zero
+      value, the lines immediately following the #if directive are
+      skipped, and examined only to determine nesting of conditionals.
+      Within these lines, no token expansion occurs, and no Fortran
+      comment lines nor source lines are made available for subsequent
+      processing by the processor. Preprocessing continues with any
```

Comment:
Do we want to specifically call out that macros in the body are not evaluated in any way, in addition to the lack of expansion and source lines? I'm thinking of things like `#error`, `#warning`, and `#pragma` which aren't really about token expansion or source lines.

Reply:
You're right. We should be clear that in the skipped lines,

- Nested conditional directives are recognized but no evaluated
- No other directives are evaluated
- No macro expansion occurs
- No directive lines, Fortran source lines, or Fortran comment lines are made available for subsequent processing

Reply:
Make sure that skipped parts act as if they aren't there, and have no effect on the preprocessing.

Reply:
Added text in today's update. This caused all the subsequent specifications on "skipping" and "participating in further preprocessing" to change. Will request further reviews.